service/history/replicatorQueueProcessor.go

```diff
@@ -179,19 +179,11 @@ func (p *replicatorQueueProcessorImpl) processSyncActivityTask(task *persistence
 
 	var startedTime *int64
 	var heartbeatTime *int64
+	scheduledTime := common.Int64Ptr(activityInfo.ScheduledTime.UnixNano())
 	if activityInfo.StartedID != common.EmptyEventID {
 		startedTime = common.Int64Ptr(activityInfo.StartedTime.UnixNano())
-
-		// int64 can only represent several hundred years of time
-		// when activity is started, the hearbeat timestamp will be empty
-		// but due the in64 limitation, the actual timestamp got is
-		// roughly 17xx year.
-		// set the heartbeat timestamp to started time if empty
-		heartbeatTime = common.Int64Ptr(activityInfo.LastHeartBeatUpdatedTime.UnixNano())
-		if *heartbeatTime < *startedTime {
-			heartbeatTime = startedTime
-		}
 	}
+	heartbeatTime = common.Int64Ptr(activityInfo.LastHeartBeatUpdatedTime.UnixNano())
```

Comment:
plz add a comment that LastHeartBeatUpdatedTime must be valid when seeing the sync activity replication task